src_py/sprite.py

```diff
@@ -226,10 +226,24 @@ def layer(self):
 
         If you try to get it before it is set it will raise an attribute error.
 
+        Layer property can only be set before the sprite is added to a group,
+        after that it is read only and a sprite's layer in a group should be
+        set via the group's change_layer() method.
+
         :return: layer as an int, or raise AttributeError.
         """
         return getattr(self, '_layer')
 
+    @layer.setter
+    def layer(self, value):
+        if not self.alive():
+            setattr(self, '_layer', value)
+        else:
+            raise AttributeError("Can't set layer directly after "
```

Comment:
Did you consider changing the layer in each group here?

Something like...
```python
[group.change_layer(self, value) for group in self.groups]
```

Reply:
Yeah, I discussed it a bit [further above](https://github.com/pygame/pygame/pull/1925#issuecomment-647468318). One implementation difficulty is that not all types of groups have layers so the change_layer() method doesn't exist on those groups and will throw an error with an implementation like the above. A fairly common use case might be where people have some subset of sprites in a layered group, but also have all their sprites in one single basic mega-group for convenience. Probably we could work around that with a more complicated setter.

It's also a slightly more dramatic functionality change which I'm mostly steering clear of in these linting passes, and I guess we'd also need to update the docs to reflect that you can now change the layer of a sprite by setting it's layer. I'm sort of ambivalent on it for now, unless you think it's super useful?